src/Templates/SingleProductTemplateCompatibility.php

```diff
@@ -0,0 +1,425 @@
+				),
+			),
+			'woocommerce_before_single_product'         => array(
+				'block_name' => '',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_output_all_notices' => 10,
+				),
+			),
+			'woocommerce_before_single_product_summary' => array(
+				'block_name' => '',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_show_product_sale_flash' => 10,
+					'woocommerce_show_product_images'     => 20,
+				),
+			),
+			'woocommerce_single_product_summary'        => array(
+				'block_name' => '',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_template_single_title'   => 5,
+					'woocommerce_template_single_rating'  => 10,
+					'woocommerce_template_single_price'   => 10,
+					'woocommerce_template_single_excerpt' => 20,
+					'woocommerce_template_single_add_to_cart' => 30,
+					'woocommerce_template_single_meta'    => 40,
+					'woocommerce_template_single_sharing' => 50,
+				),
+			),
+			'woocommerce_after_single_product'          => array(
+				'block_name' => '',
+				'position'   => 'after',
+				'hooked'     => array(),
+			),
+			'woocommerce_share'                         => array(
+				'block_name' => 'woocommerce/product-details',
+				'position'   => 'before',
+				'hooked'     => array(),
+			),
+			'woocommerce_after_single_product_summary'  => array(
+				'block_name' => 'woocommerce/product-details',
+				'position'   => 'before',
+				'hooked'     => array(
+					'woocommerce_output_product_data_tabs' => 10,
+					'woocommerce_upsell_display'           => 15,
+					'woocommerce_output_related_products'  => 20,
+				),
+			),
+		);
+	}
+
+	/**
+	 * Add compatibility layer to the first and last block of the Single Product Template.
+	 *
+	 * @param string $template_content Template.
+	 * @return string
+	 */
+	public static function add_compatibility_layer( $template_content ) {
+		$wrapped_blocks = self::wrap_single_product_template( $template_content );
+		$template       = self::inject_custom_attributes_to_first_and_last_block_single_product_template( $wrapped_blocks );
+
+		return array_reduce(
+			$template,
+			function( $carry, $item ) {
+				if ( is_array( $item ) ) {
+					return $carry . serialize_blocks( $item );
+				}
+				return $carry . serialize_block( $item );
+			},
+			''
+		);
+
+	}
+
+	/**
+	 * For compatibility reason, we need to wrap the Single Product template in a div with specific class.
+	 * For more details, see https://github.com/woocommerce/woocommerce-blocks/issues/8314.
+	 *
+	 * @param string $template_content Template Content.
+	 * @return array Wrapped template content inside a div.
+	 */
+	private static function wrap_single_product_template( $template_content ) {
+		$parsed_blocks  = parse_blocks( $template_content );
+		$grouped_blocks = self::group_blocks( $parsed_blocks );
+
+		// @TODO: Check this list before terminating the Blockfied Single Product Template project.
```

Comment:
We usually use `@todo` (lowercase and no `:`) in todo comments. Example: https://github.com/woocommerce/woocommerce-blocks/blob/1e5f5a03ce90410998023b7ca00dfa367d638d50/assets/js/types/type-defs/currency.ts#L11